app/grandchallenge/algorithms/views.py

```diff
@@ -335,6 +338,82 @@ def get_context_data(self, *args, **kwargs):
         return context
 
 
+class AlgorithmImageMarkDesired(
+    LoginRequiredMixin, ObjectPermissionRequiredMixin, View
+):
+    permission_required = "algorithms.change_algorithm"
+    raise_exception = True
+
+    @cached_property
+    def algorithm(self) -> Algorithm:
+        return self.algorithm_image.algorithm
+
+    @cached_property
+    def algorithm_image(self) -> AlgorithmImage:
+        return get_object_or_404(AlgorithmImage, pk=self.kwargs["pk"])
+
+    def get_permission_object(self):
+        return self.algorithm
+
+    def post(self, request, *args, **kwargs):
```

Comment:
There is a lot of logic in the post method and its is difficult to follow. Do you think that a `FormView` would be better? Then you can use the forms validation framework. For instance, I don't think that `HttpResponseRedirect` is the correct thing to return for a validation issue, I think this should be handled in a form. All of the validation checks should be moved there.

Reply:
Hm, not super straightforward the FormView. I think I would then need a form without fields and an empty placeholder template (FormView needs a template) and would do the validation in the clean method on the form, return ValidationErrors accordingly, and then implement the `upload_to_registry_and_sagemaker` method in `form_invalid` (but only if a specific validation error was returned) and call `mark_desired_version` in `form_valid`.

Reply:
in `form_invalid` I would still need to return an HttpResponse with a message that contains either the errors or the success message that uploading to the registry is happening. Returning the form with errors, like you would usually do, doesn't work here since there is no form to return.

Reply:
You would need a form with an algorithm image right? e.g. the one that you want to mark as active.

Reply:
Maybe I'm missing something, but no matter what the form looks like, it doesn't change the fact that I need to change `form_invalid` to return the error messages properly. We don't want to display the form, we want to show the image detail page with the errors messages. So to me that means that the default behaviour of `form_invalid` needs to be overriden.

Reply:
I will make a commit to illustrate.